src/vivarium/framework/artifact/artifact.py

```diff
@@ -11,12 +10,13 @@
 archive file for convenient access and inspection.
 
 """
+from __future__ import annotations
 
 import re
 import warnings
 from collections import defaultdict
 from pathlib import Path
-from typing import Any, Dict, List, Optional, Union
+from typing import Any, Iterable, Union
```

Comment:
i think we've been leaning towards cleaning up Unions and replacing w/ | notation throughout?